### ISSUE TITLE
Adding comments to Orion 1.0 build causes Maqetta to go away

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3790]
@peller 

When I tried to add comments to a review, the Maqetta browser window went away and was replaced by the following error messages, showing in the browser where Maqetta used to be:

Error connecting to the Maqetta Server.

Please click here to return to Maqetta
Error description: error adding comment , response=RequestError: Unable to load cmd/addComment?id=b02b2c85-11ec-3090-3604-2d8b88ab8176&subject=Re%3A%20nice%20list&content=should%20have%20more%20entries&ownerId=B&email=B&next=Next&pageState=Task%20Added&pageStateList=%5B%7B%22id%22%3A%22myapp%22%2C%22xpath%22%3A%22%2Fhtml%2Fbody%5B1%5D%22%2C%22state%22%3A%22Task%20Added%22%7D%5D&viewSceneList=%7B%22DojoMobileViews%22%3A%5B%5D%7D&designerId=B&pageName=.review%2Fsnapshot%2F20130304T190230Z%2Fproject1%2Faa1.html&replyTo=b99686ac-e35c-5a6b-8efc-1ebc3f9c64ea&drawingJson=%5B%7B%22type%22%3A%22Arrow%22%2C%22x1%22%3A159%2C%22y1%22%3A202%2C%22x2%22%3A64%2C%22y2%22%3A126%2C%22colorAlias%22%3A%22B%22%2C%22commentId%22%3A%22b02b2c85-11ec-3090-3604-2d8b88ab8176%22%2C%22state%22%3A%22Task%20Added%22%2C%22stateList%22%3A%22%5B%7B%5C%22id%5C%22%3A%5C%22myapp%5C%22%2C%5C%22xpath%5C%22%3A%5C%22%2Fhtml%2Fbody%5B1%5D%5C%22%2C%5C%22state%5C%22%3A%5C%22Task%20Added%5C%22%7D%5D%22%2C%22sceneList%22%3A%22%7B%5C%22DojoMobileViews%5C%22%3A%5B%5D%7D%22%7D%5D status: 400 , reason=Error 400 The review is not added successfully. Reason:

**Comment by JonFerraiolo** [comment]
Here is the request that caused the error:

```
Request URL:http://maqetta-app-test.austin.ibm.com/maqetta/cmd/addComment?id=6650ad67-a1bd-eabf-53f0-56679e623ffe&subject=Re%3A%20nice%20buttons&content=vvv&ownerId=D&email=jon%40ferraiolo.com&next=Next&pageState=Normal&pageStateList=%5B%7B%22id%22%3A%22myapp%22%2C%22xpath%22%3A%22%2Fhtml%2Fbody%5B1%5D%22%7D%5D&viewSceneList=%7B%22DojoMobileViews%22%3A%5B%5D%7D&designerId=B&pageName=.review%2Fsnapshot%2F20130304T190230Z%2Fproject1%2Faa1.html&replyTo=cac6bcd0-bd02-9cd1-d3f9-f7ad79fe1524&drawingJson=%5B%5D
Request Method:GET
Status Code:400 The review is not added successfully. Reason:
Request Headersview source
Accept:*/*
Accept-Charset:ISO-8859-1,utf-8;q=0.7,*;q=0.3
Accept-Encoding:gzip,deflate,sdch
Accept-Language:en-US,en;q=0.8
Connection:keep-alive
Content-Type:application/x-www-form-urlencoded
Cookie:ibmSurvey=1361317932661; UnicaNIODID=dNV84CVODua-X8dTP23; __utma=101107545.1090107540.1361755191.1361755191.1361824059.2; __utmc=101107545; __utmz=101107545.1361824059.2.2.utmcsr=dojo-toolkit.33424.n3.nabble.com|utmccn=(referral)|utmcmd=referral|utmcct=/Does-dojox-mvc-group-create-databinding-td3985259.html; JSESSIONID=1e8q1s5edba8g1tswr7htymmj9
Host:HOSTNAME
Referer:http://HOSTNAME/maqetta/
User-Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2) AppleWebKit/537.22 (KHTML, like Gecko) Chrome/25.0.1364.99 Safari/537.22
X-Requested-With:XMLHttpRequest
Query String Parametersview sourceview URL encoded
id:6650ad67-a1bd-eabf-53f0-56679e623ffe
subject:Re: nice buttons
content:vvv
ownerId:D
email:jon@ferraiolo.com
next:Next
pageState:Normal
pageStateList:[{"id":"myapp","xpath":"/html/body[1]"}]
viewSceneList:{"DojoMobileViews":[]}
designerId:B
pageName:.review/snapshot/20130304T190230Z/project1/aa1.html
replyTo:cac6bcd0-bd02-9cd1-d3f9-f7ad79fe1524
drawingJson:[]
Response Headersview source
Cache-Control:must-revalidate,no-cache,no-store
Connection:keep-alive
Content-Length:1363
Content-Type:text/html;charset=ISO-8859-1
Date:Mon, 04 Mar 2013 20:38:03 GMT
Server:nginx/1.2.2
```

Response was:

```
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 400 The review is not added successfully. Reason: </title>
</head>
<body>
<h2>HTTP ERROR: 400</h2>
<p>Problem accessing /maqetta/cmd/addComment. Reason:
<pre>    The review is not added successfully. Reason: </pre></p>
<hr /><i><small>Powered by Jetty://</small></i>
```

**Comment by JonFerraiolo** [comment]
Other notes:
- It appears that the comments are successfully getting added to the server XML file
- The author is not receiving emails about the comments getting added

My guess is that the server error happens after the XML file is updated but before the emails are sent telling the author that a comment has been added.

**Comment by peller** [comment]
hopefully the error message gets passed along to the client now.

The error is caused by the e-mail, seen on the server as a CoreException, caused by:

```
javax.mail.SendFailedException: Invalid Addresses;
  nested exception is:
        com.sun.mail.smtp.SMTPAddressFailedException: 550 5.1.1 <B>... User unknown

        at com.sun.mail.smtp.SMTPTransport.rcptTo(SMTPTransport.java:1294)
        at com.sun.mail.smtp.SMTPTransport.sendMessage(SMTPTransport.java:635)
        at org.eclipse.orion.server.useradmin.UserEmailUtil.sendEmail(UserEmailUtil.java:133)
        ... 55 more
```

@jhpedemonte Is Orion configured to send e-mail to external addresses?

**Comment by jhpedemonte** [comment]
Yes, it's set up correctly.  My guess is that the R&C code is pulling in the `username`, which is now a UID such as "B", rather than the `email` to send the message.  I'm investigating.

**Comment by jhpedemonte** [comment]
Fixed by cbca412c5e550a856864207bd8053572eccd217a (new build coming up shortly).
